docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -0,0 +1,1431 @@
+   <td>Node
+   </td>
+   <td><code>python -m paddle.distributed.launch</code>
+   </td>
+   <td><a href="https://www.paddlepaddle.org.cn/documentation/docs/en/guides/06_distributed_training/cluster_quick_start_en.html">Paddle Distributed</a>
+   </td>
+  </tr>
+  <tr>
+   <td>XGBoost
+   </td>
+   <td>Worker
+   </td>
+   <td><em>Not Applicable</em>
+   </td>
+   <td><code>python</code> 
+   </td>
+   <td><a href="https://github.com/dmlc/xgboost/blob/a5a58102e5e82fa508514c34cd8e5f408dcfd3e1/python-package/xgboost/tracker.py#L17">Rabit Tracker</a> for c10d
+   </td>
+  </tr>
+  <tr>
+   <td>DeepSpeed
+   </td>
+   <td>Slot
+   </td>
+   <td>Node
+<p>
+<code>(--num_nodes)</code>
+   </td>
+   <td><code>deepspeed</code>
+   </td>
+   <td><a href="https://www.deepspeed.ai/getting-started/#resource-configuration-multi-node">DeepSpeed Distributed</a>
+   </td>
+  </tr>
+</table>
+
+Additionally, check [this document for the `mpirun` command](https://gist.github.com/vsoch/9ac7c4448dffe656d946edceaa58bd9e)
+for other MPI implementations: Intel MPI, MPICH, Spectrum MPI.
+
+## The TrainJob API
+
+The `TrainJob` exposes APIs that data scientist can override in `TrainingRuntime` to create training job:
+
+```golang
+type TrainJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of TrainJob.
+	Spec TrainJobSpec `json:"spec"`
+
+	// Status defines the current state of TrainJob.
+	Status TrainJobStatus `json:"status,omitempty"`
+}
+
+type TrainJobSpec struct {
+	// Reference to the Training Runtime.
+	TrainingRuntimeRef *TrainingRuntimeRef `json:"trainingRuntimeRef"`
+
+	// Parameters that data scientists can override
+	Trainer *Trainer `json:"trainer,omitempty"`
+
+	// Configuration for training dataset
+	DatasetConfig *DatasetConfig `json:"datasetConfig,omitempty"`
+
+	// Configuration for the pre-trained model and location for model output
+	ModelConfig *ModelConfig `json:"modelConfig,omitempty"`
+
+	// Custom metadata to apply for Job, JobSet, etc.
+	Labels      map[string]string `json:"labels,omitempty"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+type TrainingRuntimeRef struct {
+	// Name for the training runtime.
+	Name string `json:"name"`
+	// Namespace for the runtime.
+	// If namespace is set, TrainingRuntime is used. Otherwise, ClusterTrainingRuntime is used.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type TrainJobStatus struct {
+	// Conditions for the TrainJob
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+```
+
+This table explain rationale for each `TrainJob` parameter:
+
+<table>
+  <tr>
+   <td><strong>Parameter</strong>
+   </td>
+   <td><strong>What is it ?</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>TrainingRuntimeRef</code>
+   </td>
+   <td>Reference to the existing <code>TrainingRuntime</code> that pre-deployed by platform engineers
+   </td>
+  </tr>
+  <tr>
+   <td><code>Trainer</code>
```

Comment:
I don't think we defined above what exactly a Trainer is, so that would be useful in my opinion.

Reply:
@franciscojavierarceo I think, I am trying to explain it here: https://github.com/kubeflow/training-operator/blob/8dd9c85ce81b1ff9e1576e584e698408c56c162b/docs/proposals/2170-kubeflow-training-v2/README.md#the-trainer-api.
Do you think it is not sufficient ?

Reply:
Yeah that works, I suppose I meant that in the ordering of the document it may make sense to move the definition first so that when someone sees the tables they have that knowledge first. That's also a bit of a silly nit, so feel free to disregard. Thanks for putting this together!